src/Form.jsx

```diff
@@ -202,15 +201,24 @@ function Form({
         return newValues;
     };
 
-    const updateSessionValues = (session, fieldVals) => {
+    async function updateSessionValues(session, fieldVals) {
+        // Convert files of the format { url, path } to blobs
+        const blobs = (
```

Comment:
this function runs in more places than initial load, so this probably isn't the best place to convert s3 urls (only on initial session load, so fetchSession might make more sense). also ideally loading these files doesn't need to be blocking so the form can load first while the files download.

Reply:
it is also loading the files sequentially here, which also isn't ideal